data/mods/ssb/scripts.ts

```diff
@@ -234,7 +234,7 @@ export const Scripts: ModdedBattleScriptsData = {
 	actions: {
 		canTerastallize(pokemon) {
 			if (
-				pokemon.species.isMega || pokemon.species.isPrimal || pokemon.species.forme === "Ultra" ||
+				pokemon.terastallized || pokemon.species.isMega || pokemon.species.isPrimal || pokemon.species.forme === "Ultra" ||
```

Comment:
changing mid-set while terastallized causes a visual glitch where you have the option to terstallize, despite already doing so. This redundancy check fixes it.